server/paymentProviders/stripe/payment-intent.ts

```diff
@@ -46,11 +55,75 @@ const processOrder = async (order: typeof models.Order): Promise<void> => {
     await order.destroy({ force: true });
     throw new Error(errorMessage);
   }
-};
+}
+
+async function processRecurringOrder(order: typeof models.Order) {
+  const hostStripeAccount = await order.collective.getHostStripeAccount();
+  const host = await order.collective.getHostCollective();
+  const isPlatformRevenueDirectlyCollected = APPLICATION_FEE_INCOMPATIBLE_CURRENCIES.includes(toUpper(host.currency))
+    ? false
+    : host?.settings?.isPlatformRevenueDirectlyCollected ?? true;
+  const applicationFee = await getApplicationFee(order, host);
+  const paymentIntentParams: Stripe.PaymentIntentCreateParams = {
+    currency: order.currency,
+    amount: convertToStripeAmount(order.currency, order.totalAmount),
+    description: order.description,
+  };
+
+  if (applicationFee && isPlatformRevenueDirectlyCollected && hostStripeAccount.username !== config.stripe.accountId) {
+    // eslint-disable-next-line camelcase
+    paymentIntentParams.application_fee_amount = convertToStripeAmount(order.currency, applicationFee);
+  }
+
+  // eslint-disable-next-line camelcase
+  paymentIntentParams.payment_method_types = [order.paymentMethod?.type];
+  // eslint-disable-next-line camelcase
+  paymentIntentParams.payment_method = order.paymentMethod?.data?.stripePaymentMethodId;
+  paymentIntentParams.customer = order?.paymentMethod?.customerId;
+  paymentIntentParams.metadata = {
+    from: order.fromCollective ? `${config.host.website}/${order.fromCollective.slug}` : undefined,
+    to: `${config.host.website}/${order.collective.slug}`,
+    orderId: order.id,
+    chargeNumber: order.Subscription.chargeNumber,
+    chargeRetryCount: order.Subscription.chargeRetryCount,
+  };
+
+  try {
+    let paymentIntent = await stripe.paymentIntents.create(paymentIntentParams, {
+      stripeAccount: hostStripeAccount.username,
+    });
+
+    order.update({ data: { ...order.data, paymentIntent: { id: paymentIntent.id, status: paymentIntent.status } } });
+    await order.save();
```

Comment:
Do we really need to update order here right before confirming?

Also, `order.update` returns a Promise of the update query, you don't need to call save after it.
```suggestion
    await order.update({ data: { ...order.data, paymentIntent: { id: paymentIntent.id, status: paymentIntent.status } } });
```